monitor/statistics.hpp

```diff
@@ -28,6 +28,10 @@
 #include <mutex>
 #include <sstream>
 #include <thread>
+#include <map>
+
+#include <boost/thread/locks.hpp>
+#include <boost/thread/shared_mutex.hpp>
```

Comment:
Thank you.

As per locks - shared locks is the first sign that locking is not fine-grained, and instead of data some piece of code is guarded. It is **always** a red flag to see whether there are design problems.

Also, we can not use c++14 since Ubuntu Precise and older RHELs which we support do not have this compiler.
You said that statistics reads are rare, thus it should not be a problem.